lib/logqueue-fifo.c

```diff
@@ -267,6 +268,7 @@ log_queue_fifo_push_tail(LogQueue *s, LogMessage *msg, const LogPathOptions *pat
            * input thread finishes */
 
           main_loop_worker_register_batch_callback(&self->qoverflow_input[thread_id].cb);
+          self->qoverflow_input[thread_id].cb.user_data = log_queue_ref(s);
           self->qoverflow_input[thread_id].finish_cb_registered = TRUE;
```

Comment:
Hmm, user data is set to self here, but I don't exactly see the reason why s would be different from self.

Also is this a per message ref here? I can't see the context, but if it is, a perfect would be useful.